app/views/helpers/TimelineEvents.scala

```diff
@@ -168,7 +168,7 @@ class TimelineEvents @Inject() (
         case _: AmendmentRejected | _ => linkButton("declaration.details.fix.resubmit.button", fixAndResubmit)
       }
 
-      val cancelUrl = SubmissionController.cancelAmendment(amendmentEventAsLatest.action.decId)
+      val cancelUrl = SubmissionController.cancelAmendment(amendmentEventAsLatest.action.decId, submission.uuid)
```

Comment:
I see you are passing the SubmissionId here that the SubmissionController then sets as a session var. What I don't understand is how you can be on the timeline page without the SubmissionId already being set :thinking:

Reply:
So the submissionId is available on the timeline page, and it is used to form the timeline as a parameter to `TimelineEvents.apply(submission)`, here I am using that parameter in the formation of the url.

Reply:
But I need the submissionId on the legal declaration page, where previously it was not available.